src/Avalonia.Base/Media/Fonts/Tables/VerticalHeaderTable.cs

```diff
@@ -0,0 +1,127 @@
+namespace Avalonia.Media.Fonts.Tables
+{
+    internal readonly struct VerticalHeaderTable
+    {
+        internal const string TableName = "vhea";
+        internal static OpenTypeTag Tag { get; } = OpenTypeTag.Parse(TableName);
+
+        public VerticalHeaderTable(
+            short ascender,
+            short descender,
+            short lineGap,
+            ushort advanceHeightMax,
+            short minTopSideBearing,
+            short minBottomSideBearing,
+            short yMaxExtent,
+            short caretSlopeRise,
+            short caretSlopeRun,
+            short caretOffset,
+            ushort numberOfVMetrics)
+        {
+            Ascender = ascender;
+            Descender = descender;
+            LineGap = lineGap;
+            AdvanceHeightMax = advanceHeightMax;
+            MinTopSideBearing = minTopSideBearing;
+            MinBottomSideBearing = minBottomSideBearing;
+            YMaxExtent = yMaxExtent;
+            CaretSlopeRise = caretSlopeRise;
+            CaretSlopeRun = caretSlopeRun;
+            CaretOffset = caretOffset;
+            NumberOfVMetrics = numberOfVMetrics;
+        }
+
+        public ushort AdvanceHeightMax { get; }
+
+        public short Ascender { get; }
+
+        public short CaretOffset { get; }
+
+        public short CaretSlopeRise { get; }
+
+        public short CaretSlopeRun { get; }
+
+        public short Descender { get; }
+
+        public short LineGap { get; }
+
+        public short MinTopSideBearing { get; }
+
+        public short MinBottomSideBearing { get; }
+
+        public ushort NumberOfVMetrics { get; }
+
+        public short YMaxExtent { get; }
+
+        public static VerticalHeaderTable? Load(IGlyphTypeface fontFace)
+        {
+            if (!fontFace.PlatformTypeface.TryGetTable(Tag, out var table))
+            {
+                return null;
+            }
+
+            var binaryReader = new BigEndianBinaryReader(table.Span);
+
+            return Load(binaryReader);
+        }
+
+        private static VerticalHeaderTable Load(BigEndianBinaryReader reader)
+        {
+            // See OpenType spec for vhea:
+            // | Fixed  | version             | 0x00010000 (1.0)                                                                |
```

Comment:
The version of a [vhea table](https://learn.microsoft.com/en-us/typography/opentype/spec/vhea) is a `Version16Dot16`, not a `Fixed`. Same size, but different values (and also different from two int16 fields).